lib/Net/SAML2/Binding/POST.pm

```diff
@@ -76,4 +81,47 @@ sub handle_response {
     return $xml;
 }
 
+=head2 prepare_request( $request )
+
+Sign and encode the SAMLRequest.
+
+=cut
+
+sub prepare_request {
+    my ($self, $request) = @_;
+
```

Comment:
`if has_key && has_cert` otherwise we cannot sign/prepare the request. I think this method should also be named `sign_request`.

Reply:
I "struggled" with the name and settleed as is because it also base64 encodes it.  I am not really attached to it though.